src/lib/operators/aggregate_hash.cpp

```diff
@@ -227,8 +241,9 @@ __attribute__((hot)) void AggregateHash::_aggregate_segment(ChunkID chunk_id, Co
   };
 
   // If we have more than one aggregate function (and thus more than one context), it makes sense to cache the results
-  // indexes, see get_or_add_result for details.
-  if (_contexts_per_column.size() > 1) {
+  // indexes, see get_or_add_result for details. Furthermore, if we use the immediate key shortcut, we need to pass
+  //  true_type so that the aggregate keys are checked for immediate access values.
+  if (_contexts_per_column.size() > 1 || _use_immediate_key_shortcut) {
```

Comment:
Kind of surprising to see that it's the same path for caching and immediate key shortcut. I assume caching is no longer required with "perfect hashing", right?

Reply:
Correct. Changed it to

> Furthermore, if we use the immediate key shortcut (which uses the same code path as caching), we need to pass true_type so that the aggregate keys are checked for immediate access values.

Any idea how to further improve this?

Reply:
Maybe start by saying what is being done (pass true type for certain optimizations, which has a path that checks for a mask ... blah ...; two cases where we use this path ...) ?

Reply:
Added "Pass true_type into get_or_add_result to enable certain optimizations". I do not want to go into more detail here - that level of detail will be found in get_or_add_result.